### PR TITLE
Adding happy path tests for payload endpoints with DB

### PR DESCRIPTION
**Pull request by tahmidefaz**
## What?
Happy path test for the payloads and request_id payloads endpoints that makes use of the DB.

## Why?
More confident payload tracker releases.
These tests cover almost the whole payload endpoints API code, starting with a mocked API request all the way to fetching and validating data from an actual database.
If these tests were to fail, it could mean something is terribly wrong with the `/payloads` and `/payloads/{request_id}` endpoints.

## How?
* Some slight refactoring to enable us to swap out the DB objects of the API with the one from the tests.
* Refactoring to move the `MakeTestRequest` function for the sake of code reusability.
* Adding 2 tests for the endpoints that make use of the DB.

## Testing
* 2 happy path test for the `/payloads` and `/payloads/{request_id}` endpoints.

## Anything Else?
Any other notes about the PR that would be useful for the reviewer. 

## Secure Coding Practices Checklist Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist
- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [ ] General Coding Practices

### CODE REVIEW

**Review by SteveHNH — APPROVED** [review]
lgtm

File: internal/endpoints/endpoints_db_test.go/endpoints_db_suite_test.go [hunk 1]
```diff
@@ -0,0 +1,15 @@
+package endpoints_db_test
```
Comment (dagbay-rh) [inline internal/endpoints/endpoints_db_test.go/endpoints_db_suite_test.go:1]: nit pick: the parent directory of this file having `.go` at the end looks a little strange to me

**Review by aleccohan — APPROVED** [review]
LGTM!

File: internal/endpoints/endpoints_db_test.go/endpoints_db_suite_test.go (hunk 1 above)
Comment (SteveHNH) [reply internal/endpoints/endpoints_db_test.go/endpoints_db_suite_test.go:1]: This is a good catch. i didn't notice it.

Comment (tahmidefaz) [reply internal/endpoints/endpoints_db_test.go/endpoints_db_suite_test.go]: Done! :100:

**Review by dagbay-rh — APPROVED** [review]

**Pull request merged**